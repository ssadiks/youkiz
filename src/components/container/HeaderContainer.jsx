import React from 'react';
import { Link } from 'react-router-dom';

const HeaderContainer = () => (
  <div>
    <ul>
      <li><Link to="/">Home</Link></li>
      <li><Link to="/videos">Videos</Link></li>
      <li><Link to="/topics">not found</Link></li>
    </ul>
  </div>
);

export default HeaderContainer;
