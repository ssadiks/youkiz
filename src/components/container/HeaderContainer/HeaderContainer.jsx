import React from 'react';
import { Link } from 'react-router-dom';
import LanguageContainer from '../LanguageContainer/LanguageContainer';

const HeaderContainer = () => (
  <div>
    <ul>
      <li><Link to="/">Home</Link></li>
      <li><Link to="/videos">Videos</Link></li>
      <li><Link to="/bo">Back office</Link></li>
      <li><Link to="/topics">not found</Link></li>
    </ul>
    <LanguageContainer />
  </div>
);

export default HeaderContainer;
