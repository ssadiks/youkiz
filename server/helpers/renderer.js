import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import serialize from 'serialize-javascript';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Routes from '../../src/Routes';
import HeaderContainer from '../../src/components/container/HeaderContainer/HeaderContainer';

const renderer = (req, store) => {
  const muiTheme = getMuiTheme({
    userAgent: req.headers['user-agent']
  });

  const content = renderToString(
    <Provider store={store}>
      <MuiThemeProvider muiTheme={muiTheme}>
        <StaticRouter location={req.path} context={{}}>
          <div>
            {renderRoutes(Routes)}
          </div>
        </StaticRouter>
      </MuiThemeProvider>
    </Provider>
  );

  return `
    <html>
      <head>
        <link href="css/style.css" rel="stylesheet">
      </head>
      <body>
        <div id="root">${content}</div>
        <script>
          window.INITIAL_STATE = ${serialize(store.getState())}
        </script>
        <script src="js/vendors.js"></script>
        <script src="js/bundle.js"></script>
      </body>
    </html>
  `;
};

export default renderer;
