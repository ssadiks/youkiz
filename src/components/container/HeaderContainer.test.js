import React from 'react';
import renderer from 'react-test-renderer';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import HeaderContainer from './HeaderContainer';

describe('HeaderContainer render', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Router>
        <HeaderContainer />
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
