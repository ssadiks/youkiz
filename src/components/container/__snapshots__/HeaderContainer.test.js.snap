// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer render renders correctly 1`] = `
<div>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/videos"
        onClick={[Function]}
      >
        Videos
      </a>
    </li>
    <li>
      <a
        href="/bo"
        onClick={[Function]}
      >
        Back office
      </a>
    </li>
    <li>
      <a
        href="/topics"
        onClick={[Function]}
      >
        not found
      </a>
    </li>
  </ul>
</div>
`;
